### PR TITLE
Fix extension for 10+ Refactored to reduce built index.js from 200kb to 10kb

### PR DESCRIPTION
**Pull request by u12206050**

### CODE REVIEW

**Comment by hanneskuettner** [comment]
Thank you for this contribution. I'll try to get around to reviewing on monday.

**Comment by madc** [comment]
Wow, great PR, @u12206050!

**Comment by u12206050** [comment]
Any chance getting this published today? :) (Probably as a new major version then those still using v9 won't suddenly get this version)

**Comment by madc** [comment]
@u12206050
When trying to manually build this extension with `yarn build` it fails with `directus-extension: command not found`.
Adding `@directus/extensions-sdk` as a dev dependency seems to solve this. Do you think, it makes sense to add it?

**Comment by u12206050** [comment]
Yes of course, sorry strange I forgot that. I actually built it with pnpm, so might be I had it installed globally.

**Comment by u12206050** [comment]
Have added it to dev deps.

**Comment by madc** [comment]
I've been using you PR on a Project for a week now and it works like a charm!

**Comment by madc** [comment]
On thin i noticed are the colors are hard/imposible to read and the margins are of on Directus 10:
image
image

Have you seen this? This also happens with the current published version, just wondered if you see this behavior too.

**Comment by u12206050** [comment]
Yes I have noticed that and fixed it locally by adding some custom css, but maybe we should add it to the library?

**Comment by madc** [comment]
I was planning on a PR after yours was merged. But as you already solved that, you mind sharing your workaround?

**Comment by u12206050** [comment]
Sure this is what I have.

```
body .group-tabs .tab-list {
    background: none;
}

body .group-tabs .tab-panel {
    padding: 24px 0;
}

body .group-tabs .tab-button.v-button {
    --background-normal: var(--theme--background-subdued);
	--v-button-border-color: var(--theme--border-color);
    --v-button-color: var(--theme--primary);
    --v-button-color-hover: var(--theme--primary-accent);
    --v-button-background-color: var(--theme--background);
    --v-button-background-color-hover: var(--theme--background-subdued);
}
body .group-tabs .tab-button.v-button[aria-selected='true'] {
    --v-button-background-color: transparent;
    --v-button-color: var(--theme--primary-accent);
    --v-button-background-color-hover: var(--theme--background-subdued);
    --theme--border-color: var(--theme--primary-accent);
}
    
body .group-tabs .tab-button.v-button .button {
    border: none;
    border-radius: 8px 8px 2px 2px;
    border-bottom: 2px solid var(--theme--border-color);
    padding: 16px 12px;
}
```

**Comment by madc** [comment]
image
Very nice, thanks for sharing!

**Comment by hanneskuettner** [comment]
Hi @u12206050, I'm finally able to take some time to give this extension some love again, sorry that it took this long, had to finish my thesis first!

I'd love to accept your PR, there are some minor things I'd like to change before accepting it tho. I however can't modify anything before merging since the changes are made on the main branch of your fork (at least I think that's the reason). Could you either incorporate my requested changes, or recreate the PR from a feature branch, then I can quickly do them myself while giving you the credit you deserve? Thank you! Much appreciated!

File: package-lock.json
Comment (hanneskuettner) [inline package-lock.json:1]: Please remove, this package uses pnpm and I don't want to keep the package lock around as well.

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
 	"name": "directus-extension-group-tabs-interface",
 	"icon": "extension",
-	"version": "1.2.2",
+	"version": "1.3.0",
```
Comment (hanneskuettner) [inline package.json]: The version number stays the same, semantic-release-bot handles the increase for us.

File: src/group-tabs.vue [hunk 2]
```diff
@@ -80,6 +46,23 @@ watch(
 	}
 );
 
+function isEqual(obj1: any, obj2: any) {
```
Comment (hanneskuettner) [inline src/group-tabs.vue]: I prefer to keep using isEqual from lodash, no need to implement it ourselves. Right now we implicitly depend on the lodash that is bundled with directus/app, it is probably a good idea to add `lodash-es` to the dev dependencies. They bundle should still stay small.

File: src/index.ts [hunk 3]
```diff
@@ -1,7 +1,6 @@
-import { defineInterface } from '@directus/utils';
 import InterfaceGroupTabs from './group-tabs.vue';
 
-export default defineInterface({
+export default {
```
Comment (hanneskuettner) [inline src/index.ts:3]: Whats the reason for removing this? As far as I can see `defineInterface` is available in the `@directus/extension-sdk` and I think it should be kept.

**Review by hanneskuettner — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 3 above)
Comment (u12206050) [reply src/index.ts:3]: This is actually the main reason for what causes such a large bundle size. I have raised the issue with Directus and they have been looking into it. Will probably be fixed in an upcoming release. Until then without it still works.

File: package-lock.json
Comment (u12206050) [reply package-lock.json:1]: Done.

**Comment by u12206050** [comment]
> Hi @u12206050, I'm finally able to take some time to give this extension some love again, sorry that it took this long, had to finish my thesis first!
> 
> I'd love to accept your PR, there are some minor things I'd like to change before accepting it tho. I however can't modify anything before merging since the changes are made on the main branch of your fork (at least I think that's the reason). Could you either incorporate my requested changes, or recreate the PR from a feature branch, then I can quickly do them myself while giving you the credit you deserve? Thank you! Much appreciated!

Have done all the changes as requested.

File: src/index.ts (hunk 3 above)
Comment (hanneskuettner) [reply src/index.ts:3]: Good to know! Thanks for raising that issue.

Comment (hanneskuettner) [reply src/index.ts:3]: Hmmmm, after trying both versions (with and without defineInterface) locally I don't see any difference in bundle size for me. But I'll accept it like this and look into it after the fact.

**Review by hanneskuettner — APPROVED** [review]

**Comment by hanneskuettner** [comment]
Thank you again!

**Pull request merged**